filament/backend/src/vulkan/VulkanDriver.cpp

```diff
@@ -1237,11 +1272,26 @@ void VulkanDriver::draw(PipelineState pipelineState, Handle<HwRenderPrimitive> r
     vkCmdDrawIndexed(cmdbuffer, indexCount, instanceCount, firstIndex, vertexOffset, firstInstId);
 }
 
-
 void VulkanDriver::beginTimerQuery(Handle<HwTimerQuery> tqh) {
+    VulkanCommandBuffer* commands = mContext.currentCommands;
+    ASSERT_POSTCONDITION(commands, "Timer queries can occur only within a beginFrame / endFrame.");
+
+    VulkanTimerQuery* vtq = handle_cast<VulkanTimerQuery>(mHandleMap, tqh);
+    const uint32_t index = vtq->startingQueryIndex;
+    const VkPipelineStageFlagBits stage = VK_PIPELINE_STAGE_TOP_OF_PIPE_BIT;
```

Comment:
I think you might need `VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT` here as well. 
i.e. you want to take a timestamp after everything before this call has been executed on the GPU.

It seems to me that using `TOP` here would means that you measure from when the CPU gets to this command.

Reply:
You're right, both should be BOTTOM.  Fixed.  (Btw I think `TOP` has nothing to do with CPU, it's more about the when the command is initially consumed by the GPU.)